commander/src/commands/transaction/sign.ts

```diff
@@ -71,14 +103,19 @@ export default class SignCommand extends BaseCommand {
 			flags: {
 				networkIdentifier: networkIdentifierSource,
 				passphrase: passphraseSource,
+				'mandatory-key': mandatoryKeys,
+				'optional-key': optionalKeys,
+				'number-of-signatures': numberOfSignatures,
+				'number-of-passphrases': numberOfPassphrases,
 			},
 		} = this.parse(SignCommand);
 
 		const { transaction }: Args = args;
 		const transactionInput = transaction || (await getTransactionInput());
 		const transactionObject = parseTransactionString(transactionInput);
 		const passphrase =
-			passphraseSource ?? (await getPassphraseFromPrompt('passphrase', true));
+			passphraseSource ??
+			(await getPassphrasesFromPrompt(numberOfPassphrases as number));
```

Comment:
flag comes in always as string, so it should `parseInt`

Reply:
in this case the flag is `flagParser.integer({` i am casting it to number as the flag can resolve to `number|undefined` so

Reply:
if that's the case, i think it should be a problem with the typescript as `getPassphrasesFromPrompt` has default value, and it should accept the `undefined`